pkg/virt-config/feature-gates.go

```diff
@@ -28,18 +28,19 @@ const (
 	IgnitionGate      = "ExperimentalIgnitionSupport"
 	LiveMigrationGate = "LiveMigration"
 	// SRIOVLiveMigrationGate enable's Live Migration for VM's with SRIOV interfaces.
-	SRIOVLiveMigrationGate     = "SRIOVLiveMigration"
-	CPUNodeDiscoveryGate       = "CPUNodeDiscovery"
-	HypervStrictCheckGate      = "HypervStrictCheck"
-	SidecarGate                = "Sidecar"
-	GPUGate                    = "GPU"
-	HostDevicesGate            = "HostDevices"
-	SnapshotGate               = "Snapshot"
-	HotplugVolumesGate         = "HotplugVolumes"
-	HostDiskGate               = "HostDisk"
-	VirtIOFSGate               = "ExperimentalVirtiofsSupport"
-	MacvtapGate                = "Macvtap"
-	DownwardMetricsFeatureGate = "DownwardMetrics"
+	SRIOVLiveMigrationGate            = "SRIOVLiveMigration"
+	CPUNodeDiscoveryGate              = "CPUNodeDiscovery"
+	HypervStrictCheckGate             = "HypervStrictCheck"
+	SidecarGate                       = "Sidecar"
+	GPUGate                           = "GPU"
+	HostDevicesGate                   = "HostDevices"
+	SnapshotGate                      = "Snapshot"
+	HotplugVolumesGate                = "HotplugVolumes"
+	HostDiskGate                      = "HostDisk"
+	VirtIOFSGate                      = "ExperimentalVirtiofsSupport"
+	MacvtapGate                       = "Macvtap"
+	DownwardMetricsFeatureGate        = "DownwardMetrics"
+	HostModelCPUMigratableFeatureGate = "HostModelCPUModeMigratable"
```

Comment:
Does this feature gate disable existing behavior? For example, if someone has a cluster and is migrating VMIs today using the default cpu model, that model is 'host-model' right? So when they update, migrations would no longer work for them until they enable this feature gate which then grants them the previous behavior.

Rather than opting into HostModel migration, since that was the previous default behavior, I'd instead expect a feature gate that allows people to opt-out of host model migration... that's if we need a feature gate at all. I'd be fine with no feature gate.

Reply:
On the one hand I think it makes sense to have this feature gate - in order for VMIs with host-model mode to migrate there's a need for a special care (need to make sure node-labeller runs and that there are nodes in the cluster that do support the same model). Also, HCO uses a similar feature gate for the same reason, as @vladikr [stated above](https://github.com/kubevirt/kubevirt/pull/5623#pullrequestreview-668325105).

On the other hand I understand your point. Backward comparability can be hurt because of this.

The question, IMO, is what's worse:
1. Someone updates Kubevirt and fails to migrate because a new feature gate is turned off by default. In such a case, to fix the problem, the user has to turn on the feature gate.
2. The user updates Kubevirt and everything works as expected. Sometime in the future the system admin adds a new node (or many nodes) that does not support the other nodes' CPU model. The migrations will succeed, but no VM would every migrate to the new node from the old one (and vice-versa). The system administrator might not even notice this for a while and when he does he will probably have to work pretty hard to understand why this is the behavior.

In other words, as I see it, the question is if we want to force our users to raise awareness about this or not. We can force every user that uses such migrations to explicitly declare he understands the requirements, or we can keep things as they are, add documentation, and expect users to read and follow it.

What do you think about this? @vladikr

Reply:
for the record, i'd be fine with no feature gate.  The node labeller is part of the kubevirt core, so in the 99% case, everyone will have it working properly.

If anything, as an escape hatch we could add a feature gate that disables the new target node host-model cpu scheduling in the event someone needs to work around something unique in their environment. something like `DisableHostModelCPUScheduling`

Reply:
The reason why I was in favor of this feature gate is simply that we could quickly reject migrations of VMIs with host-model in the case when the chances for a successful migration are low.
In this case, the user will be able to "correct" the issue by starting the VMI with a different, migratable, CPU model.
This would also prevent multiple migration re-tries during upgrades. 

However, with the current approach, we will not attempt to migrate the VMI if there are no host-model translation labels on the node, so I think we don't need the additional feature gate.

Reply:
Feature gate is now removed.